draft-ietf-quic-transport.md

```diff
@@ -1198,12 +1198,91 @@ The fields of a GOAWAY frame are as follows:
-size is 1350 bytes for IPv6 and 1370 bytes for IPv4.  To optimize better,
-endpoints MAY use PLPMTUD {{!RFC4821}} for detecting the path's MTU and setting
-the maximum packet size appropriately.
+The maximum packet size for QUIC is the maximum size of the entire UDP payload,
+including the public header, encrypted payload, and any authentication fields.
+All QUIC packets SHOULD be sized to fit within the path's MTU to avoid IP
+fragmentation. To optimize bandwidth efficiency, endpoints SHOULD use Path MTU
+Discovery ({{!RFC1191}}, {{!RFC1981}}) or Packetization Layer Path MTU
+Discovery ({{!RFC4821}}) for detecting the path's MTU, setting the maximum
+packet size appropriately, and storing the result of previous PMTU
+determinations. In the absence of these mechanisms,  the recommended default
+maximum packet size is 1350 bytes for IPv6 and 1370 bytes for IPv4. 
+
+(TODO: Should there be a high minimum MTU for QUIC to avoid ICMP attacks? If so,
+does the endpoint fail over to TCP or simply allow fragmentation?)
+
+QUIC endpoints MUST maintain a separate PMTU estimate for each IP address the
+peer is using in the connection. Endpoints SHOULD maintain an estimate for each
+combination of local and remote IP addresses (as each pairing may have a different
+minimum MTU in the path).
+
+## Packet MTU Determination
+
+QUIC endpoints SHOULD set the Don't Fragment (DF) bit in the IP header of selected
+QUIC datagrams. These packets MUST use PADDING frames, as necessary, to raise the
+overall packet size to the expected maximum.
+
+### DF Marking {#dfmarking}
+
+The first packet from the client MUST be maximum-size and SHOULD set the DF bit.
+
+The last server-generated packet in the Transport Handshake SHOULD be padded to
+maximum-size with the DF bit set. Earlier packets may be smaller to save server
+resources until various handshake mechanisms have validated the client.
+
+QUIC endpoints SHOULD set the DF bit on the first packet sent to or from an IP address
+new to the QUIC connection, and pad the payload appropriately.
+
+QUIC endpoints MAY set DF and pad packets when implementing probes as described in
+{{!RFC1191}}, {{!RFC1981}}, or {{!RFC4821}}.
+
+QUIC endpoints MAY set DF and pad packets when it has evidence that the path between two
+previously used IP addresses has changed.
+
+### Special Considerations for Path MTU Discovery 
+
+Traditional ICMP-based path MTU discovery ({{!RFC1191}}, {{!RFC1981}}) is
+potentially vulnerable to off-path attacks that succesfully guess the IP/port 4-tuple
+and reduce the MTU to a bandwidth-inefficient value. TCP connections mitigate this
+risk by using the (at minimum) 8 bytes of transport header echoed in the ICMP message
+to validate the TCP sequence number as valid for the current connection. However, as
+QUIC operates over UDP, this echoed information may consist only of the IP and UDP
+headers, which hold no connection-specific context except the IP/port 4-tuple.
```

Comment:
This isn't strictly correct.  There are 4 16-bit values (source port, destination port, length, checksum).  The problem is that not many implementations are willing to remember length and checksum for the packets that they sent.  Also, including length means that the packet contains very little entropy, meaning that it's relatively easy to guess from off-path.